### ISSUE TITLE
npm install from a git repo fail when semver is specified

### ISSUE THREAD

**Issue by CxRes** [npm/npm#17151]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [x] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
When I run:
```
npm install gitlab:Project/program#semver:1.0.0
```
npm install complains:
```
npm ERR!: fatal: Invalid object name 'semver'.
```

this contradicts example given in: 
https://docs.npmjs.com/cli/install

#### How can the CLI team reproduce the problem?
Run:
`npm install github:Project/program#semver:1.0.0`

### supporting information:

 - `npm -v` prints: v6.5.0
 - `node -v` prints: v3.10.3
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Windows 10

Log attached!
[npm-debug.log.txt](https://github.com/npm/npm/files/1068907/npm-debug.log.txt)

**Comment by CxRes** [comment]
Works in the latest release 
node: v8.1.0
npm: v5.0.3

**Comment by kenany** [comment]
Closing since this feature was not implemented in v3.10.3, hence this is sort of expected.
